tests/ci/run_s2n_integration.sh

```diff
@@ -1,25 +1,104 @@
-#!/bin/bash -ex
+#!/bin/bash -exu
 # Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 # SPDX-License-Identifier: Apache-2.0
 
 source tests/ci/common_posix_setup.sh
 
-AWS_LC_DIR=${PWD##*/}
+# Set up environment.
+
+# ROOT
+#  |
+#  - AWS_LC_DIR
+#    |
+#    - aws-lc
+#  |
+#  - SCRATCH_FOLDER
+#    |
+#    - s2n-tls
+#    - AWS_LC_BUILD_FOLDER
+#    - AWS_LC_INSTALL_FOLDER
+#    - S2N_TLS_BUILD_FOLDER
+
+# Assumes script is executed from the root of aws-lc directory
+AWS_LC_DIR=$(pwd)
 cd ../
 ROOT=$(pwd)
-mkdir -p aws-lc-build aws-lc-install s2n-tls-build
+
+SCRATCH_FOLDER=${ROOT}/"SCRATCH_AWSLC_S2N_INTERN_TEST"
+AWS_LC_BUILD_FOLDER="${SCRATCH_FOLDER}/aws-lc-build"
+AWS_LC_INSTALL_FOLDER="${SCRATCH_FOLDER}/aws-lc-install"
+S2N_TLS_BUILD_FOLDER="${SCRATCH_FOLDER}/s2n-tls-build"
+
+# Make script execution idempotent.
+mkdir -p ${SCRATCH_FOLDER}
+rm -rf ${SCRATCH_FOLDER}/*
+cd ${SCRATCH_FOLDER}
+
+# Test helper functions.
+# Assumes OS where the Ninja executable is named "ninja". This is not true for
```

Comment:
Jotted down a tasks to make this more general. Also need to fix `cmake` vs `cmake` on AL2...

https://github.com/awslabs/aws-lc/blob/main/tests/ci/common_posix_setup.sh#L37 handles this. But the build in that function is too specific for this script.